### DISCUSSION TITLE
Do you do custom jobs? (pyetrade)

### DISCUSSION THREAD

**Discussion by buddhahoon** [jessecooper/pyetrade#68] [Q&A]
Would you be open to doing a custom build of pyetrade?

**Comment by mw66** [comment]
What exactly you are looking for?

**Reply by buddhahoon** [reply]
Taking the pyetrade that jesse built and modding it a bit. 
Description:
Connection: To start i would want it to have a connection area, showing the output of the connection ie successful or not. Account Stats: The i would want it to automatically pull the account stats letting me know current cash available, but not include margin. I do not want the bot to trade on margin. I would also want it to display open buy's and sell's if any. Strategy: This is where i am going to set my strategy for net gains and losses. So i want to be able to set orders to automatically sell if it hits a specific %gain or sell if it hits a specific loss. The split field is how to divide the cash available. The stocks will be choosen from specific index's. The index drop down will have the indexes to choose from like dow jones, s&p 500. Then when you click select stocks it will choose X number of the stocks within that index based of the x#split field. So if 10 it would randomly select 10 stocks from that index. it would be nice if there was like a check box where i could select either at random or top 10 performers for that day. but that is not necessary. The run strat button would then place the contingent orders to buy those stocks, also setting the sell parameters for those stocks to either sell at the gain% or the loss%. The bottom section would keep track of how many positive gains or negative gains the strat is producing (successful / failure) as well as output the net gains. All of the statistics should also be outputted to a file csv or whatever is easiest that i can open in excel later to do my own statistical analysis on. lastly it needs to automatically update. so as things sell and cash is available it will run back through the same strat and place new orders. oh lastly all the buy orders need to be limits, not market. So the app should determine based of level 1 what the next sell order is in line and set the limit price accordingly. On the python example script etrade provides this should be the ask size. Essentially i just want to buy at the cheapest price available by setting the limit price based on current sell orders.

![1657172128431blob](https://user-images.githubusercontent.com/108855507/178573931-e6d55764-e41b-4395-aa2c-879b636c5ddd.png)

**Reply by mw66** [reply]
That is a full blown trading client, need lots of work.

Where your mock/screenshots coming from? does it has interface connecting to backend APIs?

**Comment by buddhahoon** [comment]
I mocked it up in python. The connection works but thats it.

**Reply by buddhahoon** [reply]
Also I barely know python, so what I have is probably crap

**Comment by mw66** [comment]
No worries, most people on github do this in their spare time.

I think it's pretty good. You can continue working on it to connect with backend APIs. Or you can publish your code, maybe other people can improve it.
